zio-http/src/main/scala/zio/http/SocketApp.scala

```diff
@@ -0,0 +1,84 @@
+package zio.http
+
+import zio._
+
+final case class SocketApp[-R](
+  handler: Handler[R, Throwable, WebSocketChannel, Any],
+  customConfig: Handler[R, Throwable, Request, Option[WebSocketConfig]],
```

Comment:
This seems too powerful. I'd prefer `Option[WebSocketConfig]`, or if that cannot be done, something like `<subset of Request> => WebSocketConfig`. 

But it feels like the config should be done at the route pattern level, e.g.:

```scala
val route = Method.GET / "connect" -> handler(socketApp.withConfig(...))
```

You already know the route and pattern there and can set config on a per-route basis. You indeed have access to the full request.

Reply:
`<subset of Request> => WebSocketConfig` would be great, especially if the result was passed into `Handler.webSocket`. As it is right now there's no way to negotiate the websocket sub-protocol for a single endpoint (such as choosing between graphql-ws and graphql-transport-ws), which would be really nice to do.